students/csimmons/lesson03/mailroom.py

```diff
@@ -0,0 +1,129 @@
+    ('Martin Acevedo', [2000, 5000]),
+    ('Sutton Keaney', [24500, 500, 3000, 5000, 1000]),
+    ('David Basilio', [750, 750, 750, 750, 5000, 750, 750]),
+    ('Andrew Laughlin', [2500, 500, 40000, 50]),
+    ('Hussein Saffouri', [1000, 1000, 2100, 7000, 55000]),
+    ]
+
+menu_prompt = '\n'.join(('Please choose from the options below:\n',
+          '1 - Send a Thank You letter',
+          '2 - Create a report',
+          '3 - Quit',
+          '>>> '))
+
+thanks_prompt = '\n'.join(('\nPlease enter a donor name:',
+                '(Enter "List" to see current donors, "Exit" to return to main menu)',
+                '\n>>>  '))
+
+gift_prompt = '\n'.join(('\nPlease enter the donation amount ("$" and commas are not needed)',
+                '>>>  '))
+
+def print_donors(donors):
+    print('\nMaster List of Donors:\n')
+    for donor in donors:
+        print(donor)
+    print('\n')
+
+def exist_donor(response, donors):
+    gift = input(gift_prompt)   
+    f_gift = float(gift)
+    f_response = response.title()
+    for i in range(len(donorlist)-1):
+        if f_response == donorlist[i][0]:
+            donorlist[i][1].append(f_gift)
+    generate_thankyou(f_response, f_gift)
+
+def new_donor(response):    
+    f_response = response.title()
+    print(('\n{} is a new donor!').format(f_response))
+    gift = input(gift_prompt)
+    f_gift = float(gift)
+    new_donor = tuple([f_response,[ ]])
+    donorlist.append(new_donor)
+    get_length = len(donorlist)-1
+    donorlist[get_length][1].append(f_gift)
+    generate_thankyou(f_response, f_gift)
+
+def print_donorlist(all_info):
+    header1 = '{:20}{:1}{:15}{:1}{:10}{:1}{:12}'.format('\n''Donor Name ', '|', ' Total Given ', '|', ' Num Gifts ', '|', ' Average Gift ')
+    header2 = ('_ ' * 32) +'\n'
+    info_row = '{dname:<20s}$ {total:>13,.2f} {gifts:^10d}  $ {avg:>12,.2f}'.format
+    print(header1)
+    print(header2)
+    for i in range(len(all_info)):
+        print(info_row(dname=all_info[i][0], total=all_info[i][1], gifts=all_info[i][2], avg=all_info[i][3]))
+    print('\n')
+
+def generate_thankyou(f_response, f_gift):
+    print("""\n
+    Dear {},
+    We would like to thank you for your extremely generous donation 
+    of ${:,.2f} to the Anonymous Charity of Seattle.\n
+    Sincerely,
+    CA Simmons \n""".format(f_response, f_gift))
+
+def send_thankyou(donorlist):
+    donors = list(map(lambda x:x[0], donorlist))
+    response = input(thanks_prompt)
+    f_response = response.title()
+    if response.lower() == 'list':
+        print_donors(donors)
+        main()
+    elif response.lower() == 'exit':
+        main()
+    elif response.title() in donors:
+        exist_donor(response, donors)
+        main()
+    else:
+        new_donor(response)
+        main()
+
+def display_report(donorlist):
+    donors = list(map(lambda x:x[0], donorlist))
+    gifts = list(map(lambda x:x[1], donorlist))
+    all_info = []
+    for i in range(len(donorlist)):
```

Comment:
here as well, no need to use range() function as you don't actually need the index. just loop over the list directly